src/transformers/models/udop/configuration_udop.py

```diff
@@ -0,0 +1,187 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+
+from ...configuration_utils import PretrainedConfig
+
+
+UDOP_PRETRAINED_CONFIG_ARCHIVE_MAP = {}
+
+
+class UdopConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`UdopDualForConditionalGeneration`] or a
+    [`UdopUnimodelForConditionalGeneration`]. It is used to instantiate a UdopDualForConditionalGeneration or
+    UdopUnimodelForConditionalGeneration model according to the specified arguments, defining the model architecture.
```

Comment:
```suggestion
    This is the configuration class to store the configuration of a [`UdopForConditionalGeneration`] or a
    [`UdopDualForConditionalGeneration`]. It is used to instantiate a UdopForConditionalGeneration or
    UdopDualForConditionalGeneration model according to the specified arguments, defining the model architecture.
```
Make let's simplify the names of the model classes a bit. Let's call the default model `UdopForConditionalGeneration` and the other variant `UdopDualForConditionalGeneration`.